app/api/v1/endpoints/strategies.py

```
@@ -234,18 +269,57 @@ async def execute_strategy(
-            credit_stmt = select(CreditAccount).where(CreditAccount.user_id == current_user.id)
+            # Use row-level locking to prevent concurrent credit overspending
+            credit_stmt = select(CreditAccount).where(CreditAccount.user_id == current_user.id).with_for_update()
             credit_result = await db.execute(credit_stmt)
             credit_account = credit_result.scalar_one_or_none()
             
             if not credit_account or credit_account.available_credits < credits_required:
                 raise HTTPException(
                     status_code=status.HTTP_402_PAYMENT_REQUIRED,
-                    detail=f"Insufficient credits. Required: {credits_required}, Available: {credit_account.available_credits if credit_account else 0}"
+                    detail=f"Strategy not owned. Purchase access first or insufficient credits. Required: {credits_required}, Available: {credit_account.available_credits if credit_account else 0}"
                 )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)